inc/zoo/swar/associative_iteration.h

```diff
@@ -69,7 +69,7 @@ template<int NB, typename B>
 constexpr auto makeLaneMaskFromMSB(SWAR<NB, B> input) {
     using S = SWAR<NB, B>;
     auto msb = input & S{S::MostSignificantBit};
-    auto msbCopiedToLSB = S{msb.value() >> (NB - 1)};
+    auto msbCopiedToLSB = S{static_cast<B>(msb.value() >> (NB - 1))};
```

Comment:
My compiler was complaining about needing casts, but I'm guessing there is probs a way to not need this?

Reply:
No, the explicit casts seem required, I don't understand myself why